plugins/woocommerce-blocks/assets/js/settings/shared/default-fields.ts

```diff
@@ -2,12 +2,26 @@
  * External dependencies
  */
 import { ComboboxControlOption } from '@woocommerce/base-components/combobox';
+import type { AllHTMLAttributes, AriaAttributes } from 'react';
 
 /**
  * Internal dependencies
  */
 import { getSetting } from './utils';
 
+// A list of attributes that can be added to a custom field when registering it.
+type CustomFieldAttributes = Pick<
+	AllHTMLAttributes< HTMLInputElement >,
+	| 'maxLength'
+	| 'disabled'
```

Comment:
Supporting disabled falls into a tricky logical state because if the field is required, disabled, and has no value, then you can't place the order, you can't see the error (disabled fields for some reason don't show validation errors), and you can't input a value.

I think we should only support disabled if the field has a saved value and that value has no errors, not sure if this can be done PHP side or JS side. Should a field that start as not disabled get disabled in the same life cycle?

Otherwise let's just remove disabled for now.

Reply:
Let's make a follow up issue for "nice to have" for this - seems like complex edge-case logic. I'll remove disabled until we figure out how we're going to handle it given the above.

Reply:
Follow up created: https://github.com/woocommerce/woocommerce/issues/43657

Reply:
Thank you!